### PR TITLE
Module 3 Week 3: Using Redux in React components

### PR DESCRIPTION
**Pull request by usamamaliq**
In this pull request, the following tasks were implemented

- Connect the app to the redux store
- Display books using useSelector
- Add books using useDispatch
- Remove books using useDispatch
- Configured categories button using useSelector & useDispatch to show current status.

Regards,
Usama Masood

### CODE REVIEW

File: src/components/BookInput.js [hunk 1]
```diff
@@ -1,8 +1,27 @@
...
+      onSubmit={(e) => {
+        e.preventDefault();
+        const title = e.target.elements.title.value;
+        const author = e.target.elements.author.value;
+        const book = { title, author, id: uuidv4() };
+        dispatch(createBook(book));
+        e.target.reset();
+      }}
+      className="BookInputContainer"
+    >
```
Comment (danmainah) [inline src/components/BookInput.js]: - [x] @usamamaliq great work on your input component, however it's not good practice to add your functionality in the return section, I would recommend you create a function and then use it in the return section to accomplish the same functionality of adding a book

File: src/components/Book.js [hunk 2]
```diff
@@ -1,12 +1,36 @@
...
+        <p className="authorName">{author}</p>
+        <button
+          id={id}
+          onClick={(e) => {
+            e.preventDefault();
+            dispatch(removeBook(id));
+          }}
+          className="removeBookButton"
+          type="button"
+        >
```
Comment (danmainah) [inline src/components/Book.js]: - [x]  @usamamaliq great work on your book component, however, it's not good practice to add your functionality in the return section, I would recommend you create a function and then use it in the return section to accomplish the same functionality of removing a book.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @usamamaliq , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by prolajumokeoni — APPROVED** [review]
Hi @usamamaliq, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀  
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**